lib/rules/prefer-const.js

```diff
@@ -240,11 +259,33 @@ module.exports = {
          * @returns {void}
          */
         function report(node) {
-            context.report({
-                node: node,
-                message: "'{{name}}' is never reassigned, use 'const' instead.",
-                data: node
-            });
+            var reportArgs = {
+                    node: node,
+                    message: "'{{name}}' is never reassigned. Use 'const' instead.",
+                    data: node
+                },
+                isNormalVariableDeclaration = node.parent.parent.parent.type === "ForInStatement" ||
+                        node.parent.parent.parent.type === "ForOfStatement" ||
+                        node.parent.parent.type === "ForStatement" ||
+                        node.parent.init,
+
+                // {let {a, b} = obj} should be written as {const {a, b} = obj}
+                isDestructuring = node.parent.parent.type === "ObjectPattern",
+
+                variableDeclarationParent = findUp(node, "VariableDeclaration");
```

Comment:
This would cause unintentional behavior.
e.g.

```js
var foo = function() {  // fixed `var foo` to `const foo`
    let a;
    ({a} = 1);
}
```

```js
var foo = function() {  // fixed `var foo` to `const foo`
    for (const b of c) { 
        let a; 
        a = 1; 
    } 
}
```

I think `findUp` should stop at `*Statement` node.
Also, the found `VariableDeclaration` node should be used for other conditions instead of `node.parent.parent`.